src/MailChimp/Facade.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Locomotive\MailChimp;
+
+use ReflectionClass;
+use Mailchimp;
+
+/**
+ * File: MailChimp Facade Class
+ *
+ * Acts as a master facade class for MailChimp to allow me to isolate and test
+ * functionality without bootstrapping the vendor API.
+ *
+ * @package Locomotive\MailChimp
+ * @link    https://bitbucket.org/mailchimp/mailchimp-api-php/ Mailchimp API Client
+ * @see     Locomotive\WordPress\Facade {
+ *    Inspired by a modified version of Kevin Sperrine's WordPress facade.
+ * }
+ */
+
+/**
+ * Class: MailChimp Facade
+ *
+ * @version 2015-02-12
+ * @since   0.0.0 (2015-02-12)
+ */
+
+class Facade
+{
+	public  static $is_initialized = false;
+	private static $__facade;
+
+	/**
+	 * MailChimp Initialization
+	 *
+	 * @version 2015-02-12
+	 * @since   2015-02-12
+	 * @access  public
+	 * @param   mixed  The arguments passed to the function
+	 * @return  void|object
+	 */
+
+	public function initialize()
+	{
+		if ( func_num_args() ) {
+
+			$reflect = new ReflectionClass( 'Mailchimp' );
```

Comment:
The `Mailchimp` class accepts two arguments: `$apikey = null`, `$opts = []`.

Won't doing `Mailchimp( func_get_args() )` just pass a single argument to the class? That single argument being an array of the arguments passed to `MailChimp\Facade::initialize()`.

That's why I implemented the `ReflectionClass` since there isn't an equivalent to `call_user_func_array()` for initiating new classes; unless `func_get_args()` has a secret feature I never noticed nor seen on StackOverflow.